src/Azure/Azure.Quantum.Client/JobManagement/Workspace.cs

```diff
@@ -21,159 +24,59 @@ namespace Microsoft.Azure.Quantum
     /// <seealso cref="Microsoft.Azure.Quantum.Client.IWorkspace" />
     public class Workspace : IWorkspace
     {
-        private readonly Uri baseUri;
-        private readonly string resourceGroupName;
-        private readonly string subscriptionId;
-        private readonly string workspaceName;
-
         /// <summary>
         /// Initializes a new instance of the <see cref="Workspace"/> class.
         /// </summary>
         /// <param name="subscriptionId">The subscription identifier.</param>
         /// <param name="resourceGroupName">Name of the resource group.</param>
         /// <param name="workspaceName">Name of the workspace.</param>
-        /// <param name="tokenCredential">The token credential.</param>
-        /// <param name="baseUri">The base URI.</param>
-        public Workspace(
-            string subscriptionId,
-            string resourceGroupName,
-            string workspaceName,
-            TokenCredential tokenCredential = null,
-            Uri baseUri = null)
-            : this(
-                subscriptionId,
-                resourceGroupName,
-                workspaceName,
-                tokenCredential == null ? null : new TokenCredentialProvider(tokenCredential),
-                baseUri)
-        {
-        }
-
-        /// <summary>
-        /// Initializes a new instance of the <see cref="Workspace"/> class.
-        /// </summary>
-        /// <param name="subscriptionId">The subscription identifier.</param>
-        /// <param name="resourceGroupName">Name of the resource group.</param>
-        /// <param name="workspaceName">Name of the workspace.</param>
-        /// <param name="location">Normalized location to use with the default endpoint.</param>
-        /// <param name="tokenCredential">The token credential.</param>
+        /// <param name="location">Azure region where the workspace was created.</param>
+        /// <param name="credential">The credentials to connect to Azure. If not provided it defaults to an interactive DefaultAzureCredentials.</param>
+        /// <param name="options">Options for the client library when communication with Azure Service..</param>
         public Workspace(
             string subscriptionId,
             string resourceGroupName,
             string workspaceName,
             string location,
-            TokenCredential tokenCredential = null)
-            : this(
-                subscriptionId,
-                resourceGroupName,
-                workspaceName,
-                tokenCredential,
-                new Uri($"https://{location}.{Constants.DefaultLocationlessEndpoint}/"))
+            TokenCredential credential = null,
```

Comment:
```suggestion
            TokenCredential credential = new DefaultAzureCredential(true),
```

Shouldn't we set `DefaultAzureCredential` as default here?

Reply:
Never mind, I see you are dealing with it in the method implementation.